spec/require_encapsulation_check.rb

```diff
@@ -0,0 +1,31 @@
+require 'English'
+REQUIRES = %w[
+  datadog/appsec
+  datadog/core
+  datadog/kit
+  datadog/profiling
+  datadog/tracing
+].freeze
+
+RSpec.describe 'require encapsulation' do
+  before(:all) do
+    # Permit Datadog::VERSION to be defined but no other constants.
+    # See the note in gemspec about requiring 'datadog/version'.
+    expect(defined?(Datadog)).to eq 'constant'
+    expect(Datadog.constants).to eq([:VERSION])
+  end
+
+  REQUIRES.each do |req|
+    context req do
+      it 'loads' do
+        pid = fork do
```

Comment:
`fork` isn't available on Windows (which we don't support.. yet. I think it's a worthy goal) but more importantly it's not available on JRuby, for which we want coverage as well.